src/browser/telemetry.js

```diff
@@ -0,0 +1,228 @@
+var _ = require('../utility');
+var urlparser = require('./url');
+
+var defaults = {
+  network: true,
+  log: true,
+  dom: true,
+  navigation: true
+};
+
+function replace(obj, name, replacement, replacements) {
+  var orig = obj[name];
+  obj[name] = replacement(orig);
+  if (replacements) {
+    replacements.push([obj, name, orig]);
+  }
+}
+
+function restore(replacements) {
+  var b;
+  while (replacements.length) {
+    b = replacements.shift();
+    b[0][b[1]] = b[2];
+  }
+}
+
+function Instrumenter(options, telemeter, rollbar, _window, _document) {
+  var autoInstrument = options.autoInstrument;
+  if (autoInstrument === false) {
+    this.autoInstrument = {};
+    return;
+  }
+  if (!_.isType(autoInstrument, 'object')) {
+    autoInstrument = defaults;
```

Comment:
The idea here is that the configuration option `autoInstrument` can be either a boolean `true | false` or an object with possible keys `network`, `log`, `dom`, `navigation`, `connectivity`. If the option is set to `false` then we do not instrument anything. If the value is set to `true` then we instrument everything based on the default values. Otherwise, we merge the object that is passed in with the defaults.

So:

```
{
    autoInstrument: false
}
=> Instrument nothing
```

```
{
    autoInstrument: true
}
<=>
{
    autoInstrument: {
        network: true,
        log: true,
        dom: true,
        navigation: true,
        connectivity: true
    }
}
```

```
{
    autoInstrument: {
        log: false,
        navigation: false
    }
}
=>
{
    autoInstrument: {
        network: true,
        log: false,
        dom: true,
        navigation: false,
        connectivity: true
    }
}
```